crates/ra_ide/src/completion/complete_record_pattern.rs

```diff
@@ -10,8 +11,24 @@ pub(super) fn complete_record_pattern(acc: &mut Completions, ctx: &CompletionCon
         _ => return,
     };
 
+    let already_present_names: Vec<SmolStr> = ctx
+        .record_lit_pat
+        .as_ref()
+        .and_then(|record_pat| record_pat.record_field_pat_list())
+        .map(|pat_list| pat_list.bind_pats())
```

Comment:
you also need to look at `record_field_pats`. `bind_pats` are only the field shorthand patterns, you're currently missing any `foo1: something`.

Reply:
Fixed and amended the test, thanks.